### PR TITLE
Suppress startup messages: created article

### LINKED ISSUE #187: Article describing how to handle (and suppress) package startup messages
I almost always want to wrap `library()` in `suppressPackageStartupMessages()` for my reprexes, but I always forget the first time around :( This is particularly true for Bioconductor packages, which tend to be very noisy because many depend on **BiocGenerics** (which masks a bunch of base function with equivalent S4 generics and reports this fact). For example, here I load the widely used **SummarizedExperiment** Bioconductor package:

``` r
library(SummarizedExperiment)
#> Loading required package: GenomicRanges
#> Loading required package: stats4
#> Loading required package: BiocGenerics
#> Loading required package: parallel
#> 
#> Attaching package: 'BiocGenerics'
#> The following objects are masked from 'package:parallel':
#> 
#>     clusterApply, clusterApplyLB, clusterCall, clusterEvalQ,
#>     clusterExport, clusterMap, parApply, parCapply, parLapply,
#>     parLapplyLB, parRapply, parSapply, parSapplyLB
#> The following objects are masked from 'package:stats':
#> 
#>     IQR, mad, sd, var, xtabs
#> The following objects are masked from 'package:base':
#> 
#>     anyDuplicated, append, as.data.frame, basename, cbind,
#>     colMeans, colnames, colSums, dirname, do.call, duplicated,
#>     eval, evalq, Filter, Find, get, grep, grepl, intersect,
#>     is.unsorted, lapply, lengths, Map, mapply, match, mget, order,
#>     paste, pmax, pmax.int, pmin, pmin.int, Position, rank, rbind,
#>     Reduce, rowMeans, rownames, rowSums, sapply, setdiff, sort,
#>     table, tapply, union, unique, unsplit, which, which.max,
#>     which.min
#> Loading required package: S4Vectors
#> 
#> Attaching package: 'S4Vectors'
#> The following object is masked from 'package:base':
#> 
#>     expand.grid
#> Loading required package: IRanges
#> Loading required package: GenomeInfoDb
#> Loading required package: Biobase
#> Welcome to Bioconductor
#> 
#>     Vignettes contain introductory material; view with
#>     'browseVignettes()'. To cite Bioconductor, see
#>     'citation("Biobase")', and for packages 'citation("pkgname")'.
#> Loading required package: DelayedArray
#> Loading required package: matrixStats
#> 
#> Attaching package: 'matrixStats'
#> The following objects are masked from 'package:Biobase':
#> 
#>     anyMissing, rowMedians
#> Loading required package: BiocParallel
#> 
#> Attaching package: 'DelayedArray'
#> The following objects are masked from 'package:matrixStats':
#> 
#>     colMaxs, colMins, colRanges, rowMaxs, rowMins, rowRanges
#> The following objects are masked from 'package:base':
#> 
#>     aperm, apply
```

Created on 2018-03-28 by the [reprex package](http://reprex.tidyverse.org) (v0.2.0).

<details>

<summary>Session info</summary>

``` r
devtools::session_info()
#> Session info -------------------------------------------------------------
#>  setting  value                                             
#>  version  R Under development (unstable) (2018-03-05 r74359)
#>  system   x86_64, darwin15.6.0                              
#>  ui       X11                                               
#>  language (EN)                                              
#>  collate  en_AU.UTF-8                                       
#>  tz       America/New_York                                  
#>  date     2018-03-28
#> Packages -----------------------------------------------------------------
#>  package              * version   date       source        
#>  backports              1.1.2     2017-12-13 CRAN (R 3.5.0)
#>  base                 * 3.5.0     2018-03-06 local         
#>  Biobase              * 2.39.2    2018-01-25 Bioconductor  
#>  BiocGenerics         * 0.25.3    2018-02-09 Bioconductor  
#>  BiocParallel         * 1.13.3    2018-03-23 Bioconductor  
#>  bitops                 1.0-6     2013-08-17 CRAN (R 3.5.0)
#>  compiler               3.5.0     2018-03-06 local         
#>  datasets             * 3.5.0     2018-03-06 local         
#>  DelayedArray         * 0.5.22    2018-03-02 Bioconductor  
#>  devtools               1.13.5    2018-02-18 CRAN (R 3.5.0)
#>  digest                 0.6.15    2018-01-28 CRAN (R 3.5.0)
#>  evaluate               0.10.1    2017-06-24 CRAN (R 3.5.0)
#>  GenomeInfoDb         * 1.15.5    2018-02-04 Bioconductor  
#>  GenomeInfoDbData       1.1.0     2017-12-16 Bioconductor  
#>  GenomicRanges        * 1.31.23   2018-03-28 Bioconductor  
#>  graphics             * 3.5.0     2018-03-06 local         
#>  grDevices            * 3.5.0     2018-03-06 local         
#>  grid                   3.5.0     2018-03-06 local         
#>  htmltools              0.3.6     2017-04-28 CRAN (R 3.5.0)
#>  IRanges              * 2.13.28   2018-02-24 Bioconductor  
#>  knitr                  1.20      2018-02-20 CRAN (R 3.5.0)
#>  lattice                0.20-35   2017-03-25 CRAN (R 3.5.0)
#>  magrittr               1.5       2014-11-22 CRAN (R 3.5.0)
#>  Matrix                 1.2-12    2017-11-20 CRAN (R 3.5.0)
#>  matrixStats          * 0.53.1    2018-02-11 CRAN (R 3.5.0)
#>  memoise                1.1.0     2017-04-21 CRAN (R 3.5.0)
#>  methods              * 3.5.0     2018-03-06 local         
#>  parallel             * 3.5.0     2018-03-06 local         
#>  Rcpp                   0.12.16   2018-03-13 CRAN (R 3.5.0)
#>  RCurl                  1.95-4.10 2018-01-04 CRAN (R 3.5.0)
#>  rmarkdown              1.9       2018-03-01 CRAN (R 3.5.0)
#>  rprojroot              1.3-2     2018-01-03 CRAN (R 3.5.0)
#>  S4Vectors            * 0.17.38   2018-03-28 Bioconductor  
#>  stats                * 3.5.0     2018-03-06 local         
#>  stats4               * 3.5.0     2018-03-06 local         
#>  stringi                1.1.7     2018-03-12 CRAN (R 3.5.0)
#>  stringr                1.3.0     2018-02-19 CRAN (R 3.5.0)
#>  SummarizedExperiment * 1.9.16    2018-03-28 Bioconductor  
#>  tools                  3.5.0     2018-03-06 local         
#>  utils                * 3.5.0     2018-03-06 local         
#>  withr                  2.1.2     2018-03-15 CRAN (R 3.5.0)
#>  XVector                0.19.9    2018-02-28 Bioconductor  
#>  yaml                   2.1.18    2018-03-08 CRAN (R 3.5.0)
#>  zlibbioc               1.25.0    2017-10-31 Bioconductor
```

</details>

Created on 2018-03-28 by the [reprex package](http://reprex.tidyverse.org) (v0.2.0).

Yeeesh ...

I see `reprex()` has a `tidyverse_quiet` argument and that there was some discussion of the more general case in https://github.com/tidyverse/reprex/issues/70. Would you be amenable to revisiting this issue? I'd be happy to lead a PR.

Thanks for a very useful package!

### PR DESCRIPTION
**Pull request by marionlouveaux**
Closes #187

Created an article to address issue #187 regarding startup messages upon loading a package.

### CODE REVIEW

File: vignettes/articles/startup_messages.Rmd [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+knitr::opts_chunk$set(
+  collapse = TRUE,
+  comment = "#>"
+)
+```
+
+When making a reprex, you might have to load libraries that are producing some messages and warnings. The goal of this article is to suppress these, without suppressing all messages and warnings of the reprex, as they might be useful for debugging. This article aims at answering the issue [#187](https://github.com/tidyverse/reprex/issues/187). 
+
+We'll review the different possibilities to silence messages and warnings when loading a library using using a fake package that gives messages and warnings upon loading.
+
```
Comment (jennybc) [inline vignettes/articles/startup_messages.Rmd]: I think you should show the techniques used right here, in one or more `eval = FALSE` chunks, for the reader who just wants to quickly see a couple of ideas and not read a longer article.

See the chunk near the top of this readxl vignette that starts "For now, here are a few ways `read_excel()` can look when you take control of the geometry:" for inspiration:

https://readxl.tidyverse.org/articles/sheet-geometry.html

Oftentimes, people just want a quick reminder of syntax they've seen before but can't remember.

File: vignettes/articles/startup_messages.Rmd [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+## Creating and loading the startuppackage
```
Comment (jennybc) [inline vignettes/articles/startup_messages.Rmd]: There is a lot of overhead associated with creating a package here in this vignette. It's a cool idea but I'm not convinced the extra work is justified. If we kept this idea, I'd have lots of comments about how we need to remove all the code about managing this package out of the vignette and into some other files.

I propose that you let go of this approach and, instead, just demonstrate startup message suppression with existing packages. The dplyr example is, in fact, the main motivator for this article, so just covering that is a tremendously helpful thing.

File: vignettes/articles/startup_messages.Rmd [hunk 3]
```diff
@@ -0,0 +1,149 @@
...
+```{r}
+detach("package:startuppackage", unload=TRUE)
+detach("package:dplyr", unload=TRUE)
+suppressPackageStartupMessages(library(startuppackage))
+```
+
+Here the `suppressPackageStartupMessages()` function suppressed all messages and warnings. 
+
+
+## Loading a library silently within a chunk
```
Comment (jennybc) [inline vignettes/articles/startup_messages.Rmd]: You need to reveal the code a user will actually use for this if they are making a reprex:

```
#+ message = FALSE, warning = FALSE
library(whatever)

#+ reprex-body
# reprex code goes here ...
```

Comment (jennybc) [reply vignettes/articles/startup_messages.Rmd]: That is: these chunk options *can* be used in an R script, you don't need to be writing R Markdown. And above I show the syntax.

File: vignettes/articles/startup_messages.Rmd [hunk 4]
```diff
@@ -0,0 +1,149 @@
...
+If the code is written in R markdown, another way to silence the warnings and messages is to use the chunk options. Make a separate chunk for loading the librairies and set the chunk options `message` and `warning` to `FALSE`. Write the rest of the reprex in a second chunk to prevent hiding useful warnings and messages for debugging.
+
+```{r, message=FALSE, warning=FALSE}
+detach("package:startuppackage", unload=TRUE)
+detach("package:dplyr", unload=TRUE)
+library(startuppackage)
+```
+
+
+## Loading the tidyverse library silently
```
Comment (jennybc) [inline vignettes/articles/startup_messages.Rmd]: This section needs to reflect that `reprex::repres()` already exposes this via the argument `tidyverse_quiet`. This defaults to `TRUE` but can be toggled in any individual call and can also be set in a user's `.Rprofile`. No one using reprex should ever be setting this option themselves "by hand".

**Comment by marionlouveaux** [comment]
Thanks a lot for your comments.
As you recommended, I removed the example startuppackage and used the dplyr package instead to simplify the article. I will probably make a separate article on my blog using the startuppackage to show all possibilities more in details.

**Comment by marionlouveaux** [comment]
I made the changes. Thank you for your comments!

**Comment by jennybc** [comment]
Thanks!

**Pull request merged**